crates/scheduler/src/call.rs

```diff
@@ -113,5 +115,9 @@ pub fn process<B: Board>(call: Api<DispatchSchedulerCall<B>>) {
 
 fn syscall<B: Board>(call: SchedulerCall<B, api::syscall::Sig>) {
     let api::syscall::Params { x1, x2, x3, x4 } = call.read();
-    call.reply(B::syscall(*x1, *x2, *x3, *x4).ok_or(Trap));
+    let result = match B::syscall(*x1, *x2, *x3, *x4) {
+        Some(result) => try { result? },
+        None => Err(Trap.into()),
+    };
+    call.reply(result);
```

Comment:
nit: `try { x?? }` is actually the implementation of `Failure:join()` I mentioned before
```suggestion
    call.reply(try { B::syscall(*x1, *x2, *x3, *x4).ok_or(Trap)?? });
```